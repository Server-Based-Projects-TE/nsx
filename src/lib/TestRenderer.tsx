import type { RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import type { ReactElement } from 'react'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'

const TestRenderer = (ui: ReactElement): RenderResult => {
  const renderResult = render(<BrowserRouter>{ui}</BrowserRouter>)
  return renderResult
}

export default TestRenderer
