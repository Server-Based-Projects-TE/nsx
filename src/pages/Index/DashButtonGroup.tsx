import React, { memo } from 'react'
import { Link } from 'react-router-dom'

import Button from '../../components/Button/Button'
import type { AdminState } from '../../redux/adminSlice'
import { logout, selectLogin } from '../../redux/adminSlice'
import { API } from '../../redux/API'
import isSuccess from '../../redux/helper/isSuccess'
import { useAppSelector } from '../../redux/hooks'
import { enqueSnackbar } from '../../redux/snackbarSlice'
import { dispatch } from '../../redux/store'

async function handleLogout(e: React.MouseEvent<HTMLButtonElement>) {
  e.preventDefault()
  const response = await dispatch(API.endpoints.logoutRequest.initiate())

  if (isSuccess(response) && 'data' in response) {
    dispatch(logout())
    dispatch(enqueSnackbar({ color: 'green', message: response.data.message }))
  }
}

const DashButtonGroup: React.FC = memo(() => {
  const login: AdminState['login'] = useAppSelector(selectLogin)
  if (login === false) return null

  return (
    <div className="flex items-center justify-around py-10">
      <Link to="/dashboard">
        <Button variant="primary" data-cy="dashoard-page-transition-link-btn">
          Dashboard
        </Button>
      </Link>
      <Button variant="secondary" data-cy="logout-btn" onClick={handleLogout}>
        Logout
      </Button>
    </div>
  )
})
DashButtonGroup.displayName = 'AdminControlPanel'

export default DashButtonGroup
