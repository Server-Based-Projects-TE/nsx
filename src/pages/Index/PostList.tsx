import React, { memo } from 'react'
import { Link } from 'react-router-dom'

import Installation from '../../components/Installation/Installation'
import Loading from '../../components/Loading/Loading'
import ButtonGroup from '../../components/Pagination/ButtonGroup'
import usePagination from '../../components/Pagination/usePagination'
import PostDate from '../../components/PostDate/PostDate'
import RTKQueryErrorMessages from '../../components/RTKQueryErrorMessages/RTKQueryErrorMessages'

const PostList: React.FC = memo(() => {
  const { page, totalPage, data, isLoading, error } = usePagination()

  if (error) {
    return <RTKQueryErrorMessages error={error} />
  }

  if (isLoading || data === undefined) {
    return <Loading />
  }

  if (data.postList.length === 0) {
    return <Installation />
  }

  return (
    <div className="flex h-full flex-col justify-between">
      <ul className="flex flex-col justify-start space-y-4">
        {data.postList?.map((post: Post, i: number) => {
          return (
            <li key={i} className="flex sm:flex-nowrap sm:space-x-2.5">
              <PostDate date={post.createdAt} />
              <div className="w-64 flex-initial break-all text-lg sm:w-auto">
                <Link
                  className="text-color-primary hover:text-gray-400"
                  to={`post/${post.id}`}
                  data-cy={`single-post-page-link-${i + 1}`}
                >
                  {post.title}
                </Link>
              </div>
            </li>
          )
        })}
      </ul>
      <ButtonGroup page={page} totalPage={totalPage} />
    </div>
  )
})
PostList.displayName = 'PostList'

export default PostList
