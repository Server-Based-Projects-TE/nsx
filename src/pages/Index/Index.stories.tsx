import type { ComponentStory } from '@storybook/react'
import React from 'react'

import Index from './'

export default {
  title: 'Pages/Index',
  component: Index,
}

const Template: ComponentStory<typeof Index> = () => <Index />

export const Default = Template.bind({})
Default.args = {}
