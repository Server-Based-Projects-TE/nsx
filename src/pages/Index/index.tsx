import React, { memo } from 'react'

import Layout from '../../components/Layout'

import AdminControlPanel from './AdminControlPanel'
import PostList from './PostList'

const Index: React.FC = memo(() => {
  return (
    <>
      <PostList />
      <AdminControlPanel />
    </>
  )
})
Index.displayName = 'Index'

const IndexPage = memo(
  () => (
    <Layout
      className="flex flex-col justify-between"
      data-cy="top-page-content-root"
    >
      <Index />
    </Layout>
  ),
  () => true
)
IndexPage.displayName = 'IndexPage'

export default IndexPage
