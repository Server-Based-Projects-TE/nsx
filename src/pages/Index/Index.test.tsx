import { waitFor } from '@testing-library/dom'
import React from 'react'

import TestRenderer from '../../lib/TestRenderer'

import Index from './'

const sleep = (ms: number): Promise<void> =>
  new Promise((resolve) => setTimeout(resolve, ms))

test('should render Index', async () => {
  const {
    container: { firstChild },
  } = TestRenderer(<Index />)

  await waitFor(
    async () => {
      await sleep(1000)
      expect(firstChild).toBeTruthy()
      expect(firstChild).toMatchSnapshot()
    },
    { timeout: 999999 }
  )
})
