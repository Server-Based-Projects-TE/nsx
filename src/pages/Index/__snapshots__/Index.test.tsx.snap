// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Index 1`] = `
<div
  class="flex flex-col justify-between w-screen h-screen"
  data-cy="top-page-content-root"
>
  <header
    class="w-full border-b border-gray-200"
    data-cy="header"
  >
    <div
      class="sm:flex-row sm:content-center container flex flex-col flex-wrap items-center h-16 gap-2 mx-auto my-4"
    >
      <a
        class="col-auto"
        data-cy="blog-title-top-page-link"
        href="/"
      >
        <h1
          class="sm:text-2xl text-xl font-bold"
        >
          Today I Learned
        </h1>
      </a>
      <p
        class="text-base text-gray-700 justify-self-start sm:pt-2.5 sm:pl-1.5"
      >
        What Today I Learned
        <span
          aria-label="note emoji"
          class="pl-1"
          role="img"
        >
          üìù
        </span>
      </p>
      <div
        class="sm:flex-grow sm:justify-end flex space-x-2"
      >
        <a
          href="/about"
        >
          <div
            class="hover:text-gray-300 text-lg"
          >
            about
          </div>
        </a>
        <a
          href="https://github.com/laststance/nsx"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="inline w-6 h-6 text-gray-500"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <title>
              Github
            </title>
            <path
              clip-rule="evenodd"
              d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </div>
  </header>
  <main
    class="container mx-auto flex-grow px-4 py-4 flex flex-col justify-between"
  >
    <div
      class="flex flex-col justify-between h-full"
    >
      <ul
        class="flex flex-col justify-start space-y-4"
      >
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            01/14/22
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-1"
              href="/post/64"
            >
              What is matter my $500 for
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/29/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-2"
              href="/post/63"
            >
              Please MUI Components all argTypes Json generator
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/27/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-3"
              href="/post/62"
            >
              Fully Design Token Styling
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/27/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-4"
              href="/post/61"
            >
              figma cource 7 hours
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/25/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-5"
              href="/post/60"
            >
              Override interface properly type
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/24/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-6"
              href="/post/59"
            >
              I have no religion
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/24/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-7"
              href="/post/58"
            >
              zzzzsb
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/21/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-8"
              href="/post/57"
            >
              validation
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/17/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-9"
              href="/post/56"
            >
              monorepo?
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/16/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-10"
              href="/post/55"
            >
              migrated vite!
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/09/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-11"
              href="/post/54"
            >
              Making Tailwind CSS + React-Hook-Form &lt;Input /&gt;
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/09/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-12"
              href="/post/53"
            >
              I was challenged reusable input component with React Hook Form
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            12/07/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-13"
              href="/post/52"
            >
              superstruct
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            11/30/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-14"
              href="/post/51"
            >
              I'm tired as a OSS fan boy
            </a>
          </div>
        </li>
        <li
          class="flex sm:flex-nowrap sm:space-x-2.5"
        >
          <div
            class="flex-initial w-24 text-lg text-center text-gray-500"
          >
            11/26/21
          </div>
          <div
            class="sm:w-auto flex-initial w-64 text-lg break-all"
          >
            <a
              class="hover:text-gray-400"
              data-cy="single-post-page-link-15"
              href="/post/49"
            >
              Hide error within Redux-Toolkit and Redux-persist combination 
            </a>
          </div>
        </li>
      </ul>
      <div
        class="flex items-center justify-center p-8 px-10 space-x-4"
      >
        <button
          class="w-14 disabled:opacity-30 disabled:cursor-not-allowed focus:outline-none hover:text-gray-400 hover:border-gray-200 hover:bg-gray-100 flex items-center justify-center h-10 border border-gray-500 rounded"
          data-cy="prev-page-btn"
          disabled=""
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 16l-4-4m0 0l4-4m-4 4h18"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          data-cy="page-count"
        >
          1
           / 
          4
        </div>
        <button
          class="w-14 disabled:opacity-30 disabled:cursor-not-allowed focus:outline-none hover:text-gray-400 hover:border-gray-200 hover:bg-gray-100 flex items-center justify-center h-10 border border-gray-500 rounded"
          data-cy="next-page-btn"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 8l4 4m0 0l-4 4m4-4H3"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="flex items-center justify-around py-10"
    >
      <a
        href="/login"
      >
        <button
          class="flex justify-center items-center shadow rounded disabled:opacity-70 disabled:cursor-not-allowed focus:outline-none font-semibold bg-green-500 hover:bg-green-600 text-white py-2 px-6 text-md"
          data-cy="login-btn"
          type="button"
        >
          <span
            class="mx-2"
          >
            Login
          </span>
        </button>
      </a>
      <a
        href="/signup"
      >
        <button
          class="flex justify-center items-center shadow rounded disabled:opacity-70 disabled:cursor-not-allowed focus:outline-none font-semibold bg-blue-500 hover:bg-blue-600 text-white py-2 px-6 text-md"
          data-cy="signup-btn"
          type="button"
        >
          <span
            class="mx-2"
          >
            Sigunup
          </span>
        </button>
      </a>
    </div>
  </main>
  <footer
    class="w-full"
  >
    <div
      class="flex flex-col items-center w-full px-6 border-t border-gray-200"
    >
      <div
        class="sm:w-2/3 py-6 text-center"
      >
        <p
          class="text-sm text-gray-600"
        >
          ¬© 2022 by Ryota Murakami
        </p>
      </div>
    </div>
  </footer>
</div>
`;
